### PR TITLE
Refactoring c# projects

### PR DESCRIPTION
**Pull request by StefH**
- [x] I acknowledge that this PR is not a solution to the Gilded Rose Kata, but an improvement to the template.

## Please provide your PR description below this line
- Use file-scoped namespaces everywhere
- Changed csproj files : added RootNamespace
- Removed `NUnit/GildedRoseTests/TextTestFixture.cs` because the same logic is defined in `csharp.NUnit/GildedRose/Program.cs`
- Program.cs is the same for both (⭐: actually I think it's even better to only have 1 main GildedRose project which is defined in the NUnit project and referenced in the other xUnit project)
- small changes from `GildedRose(IList<Item> Items)` to `GildedRose(IList<Item> items)` and `IList<Item> Items;` to `private readonly IList<Item> _items;` to be more conformant with C# .NET coding guidelines.

### CODE REVIEW

File: csharp.NUnit/GildedRose/Program.cs [hunk 1]
```diff
@@ -34,19 +34,15 @@ public static void Main(string[] args)
...
         var app = new GildedRose(items);
 
-        int days = 2;
-        if (args.Length > 0)
-        {
-            days = int.Parse(args[0]) + 1;
-        }
 
-        for (var i = 0; i < days; i++)
+        for (var i = 0; i < 31; i++)
```
Comment (codecop) [inline csharp.NUnit/GildedRose/Program.cs:45]: We need dynamic days as all other texttest fixture files have them. Please keep that

File: csharp.xUnit/GildedRose/GildedRose.cs [hunk 2]
```diff
@@ -1,89 +1,89 @@
 ﻿using System.Collections.Generic;
 
-namespace GildedRoseKata
+namespace GildedRoseKata;
+
+public class GildedRose
 {
-    public class GildedRose
+    private readonly IList<Item> _items;
```
Comment (codecop) [inline csharp.xUnit/GildedRose/GildedRose.cs:7]: Items are package access and mutable in other languages. Please do not change this. Also keep original name please

File: csharp.NUnit/GildedRose/Program.cs [hunk 3]
```diff
@@ -34,19 +34,15 @@ public static void Main(string[] args)
                 SellIn = 5,
                 Quality = 49
             },
+
```
Comment (codecop) [inline csharp.NUnit/GildedRose/Program.cs:37]: Why?

File: csharp.xUnit/GildedRose/Program.cs [hunk 4]
```diff
@@ -1,56 +1,57 @@
...
+            // this conjured item does not work properly yet
+            new Item {Name = "Conjured Mana Cake", SellIn = 3, Quality = 6}
+        };
 
-            var app = new GildedRose(Items);
+        var app = new GildedRose(items);
 
 
-            for (var i = 0; i < 31; i++)
+        for (var i = 0; i < 31; i++)
```
Comment (codecop) [inline csharp.xUnit/GildedRose/Program.cs:45]: Should be dynamic like other one

**Review by codecop — CHANGES_REQUESTED** [review]
Added some in line comments. Also please keep formatting changes minimal if possible (unless missing formatting). Thank you

File: csharp.xUnit/GildedRose/GildedRose.cs (hunk 2 above)
Comment (codecop) [reply csharp.xUnit/GildedRose/GildedRose.cs:7]: This is already part of exercise

**Comment by StefH** [comment]
@codecop 
Please check the files again if it's ok.

**Comment by codecop** [comment]
Thank you. Now I see, the original code was already inconsistent. That is a pity. So as background for you, the code should look the same as much as possible for both projects. From what I see here I think project files, GildedRose and Item are identical. Both `Program` are almost identical besides the missing dynamic `days` in the second one. While it has not been there if you can please add it to improve the consistency between the versions. Also both Approvaltests should look as similar as possible. Because you have the project open, maybe you can run a diff between both folders to verify my assumption. If all is good, then please squash your changes to one commit. We are nearly there ;-) Thank you for your patience.

**Comment by StefH** [comment]
@codecop 
- I've fixed the dynamic `days` and the two main projects are now identical. So only 1 main GildedRose project is needed. I can make that change in this PR also if you approve.
- Both Approvaltests look as similar as possible
- I did a diff between the 2 folders
- squash your changes to one commit --> Normally I don't do this, I do this when I merge the PR to master, in that case I use Squash commit, this makes the Git Timeline even cleaner.

My project
image

This project
image

**Comment by StefH** [comment]
@codecop 
Now I just see that you have already merged my PR?

**Comment by emilybache** [comment]
Hi, thanks to both of you for doing this, I'm happy to see these changes. I think I will ask @codecop to take another look and do the merge as he sees fit.

**Pull request merged**